packages/core/admin/admin/src/hooks/useEnterprise/tests/useEnterprise.test.js

```diff
@@ -0,0 +1,80 @@
+import { act, renderHook } from '@testing-library/react';
+
+import { useEnterprise } from '../useEnterprise';
+
+const CE_DATA_FIXTURE = ['CE'];
+const EE_DATA_FIXTURE = ['EE'];
+
+function setup(...args) {
+  return renderHook(() => useEnterprise(...args));
+}
+
+describe('useEnterprise (CE)', () => {
+  beforeAll(() => {
+    window.strapi.isEE = false;
+  });
+
+  test('Returns CE data', async () => {
+    const { result } = setup(CE_DATA_FIXTURE, async () => EE_DATA_FIXTURE);
+
+    expect(result.current).toBe(CE_DATA_FIXTURE);
+  });
+});
+
+describe('useEnterprise (EE)', () => {
+  beforeAll(() => {
+    window.strapi.isEE = true;
+  });
+
+  afterAll(() => {
+    window.strapi.isEE = false;
+  });
+
+  test('Returns default data on first render and EE data on second', async () => {
+    const { result, rerender } = setup(CE_DATA_FIXTURE, async () => EE_DATA_FIXTURE);
+
+    await act(async () => {
+      expect(result.current).toBe(null);
+    });
+
+    rerender();
+
+    expect(result.current).toBe(EE_DATA_FIXTURE);
+  });
```

Comment:
I find this test strange, how come it doesn't cause a render and you have to manually invoke rerender? 🤔